examples/protonect/src/registration.cpp

```diff
@@ -85,6 +90,16 @@ void Registration::apply( int dx, int dy, float dz, float& cx, float &cy) const
   cx = rx * color.fx + color.cx;
 }
 
+/**
+ * Undistort/register a whole image.
+ * @param rgb RGB color image (1920x1080).
+ * @param depth Depth image (512x424)
+ * @param [out] undistorted Undistorted depth image.
+ * @param [out] registered Image Color image fixed for depth.
+ * @param enable_filter Whether or not to drop 'too noisy' pixels.
+ * @param [out] bigdepth If not \c NULL, more accurate depth.
+ * @note \a bigdepth has some type and size requirements that are blindly assumed.
+ */
```

Comment:
By default, this function maps color pixels onto depth data, so you get a 512x424 color image. If the `bigdepth` buffer is provided, it also performs the inverse mapping. This means the buffer is filled with depth pixels mapped onto color data, so this gives you a 1920x1082 (yes, 1082) depth image with one blank row on the top and the bottom (i.e. (float*)(bigdepth->data)[1920] is the offset of the first color-mapped depth pixel).